--- plait-draw.mjs.backup	2026-02-21 17:17:17
+++ plait-draw.mjs	2026-02-21 17:26:05
@@ -1783,7 +1783,7 @@
                 .map((command) => `A ${command.rx} ${command.ry} ${command.xAxisRotation} ${command.largeArcFlag} ${command.sweepFlag} ${command.endX} ${command.endY}`)
                 .join('\n') +
             ' Z';
-        const svgElement = rs.path(pathData, { ...options, fillStyle: 'solid' });
+        const svgElement = rs.path(pathData, options);
         setPathStrokeLinecap(svgElement, 'round');
         return svgElement;
     },
@@ -2185,8 +2185,7 @@
         const maskG = drawShape(board, activeRectangle, shape, {
             stroke: SELECTION_BORDER_COLOR,
             strokeWidth: 0,
-            fill: isClosedDrawElement(element) ? SELECTION_FILL_COLOR : DefaultDrawStyle.fill,
-            fillStyle: 'solid'
+            fill: isClosedDrawElement(element) ? SELECTION_FILL_COLOR : DefaultDrawStyle.fill
         }, drawOptions);
         g.appendChild(maskG);
     }
@@ -2196,8 +2195,7 @@
             const circleG = drawCircle(PlaitBoard.getRoughSVG(board), point, 8, {
                 stroke: SELECTION_BORDER_COLOR,
                 strokeWidth: ACTIVE_STROKE_WIDTH,
-                fill: '#FFF',
-                fillStyle: 'solid'
+                fill: '#FFF'
             });
             g.appendChild(circleG);
         });
@@ -2270,7 +2268,8 @@
             stroke: strokeColor,
             strokeWidth,
             fill,
-            strokeLineDash
+            strokeLineDash,
+            fillStyle: element.fillStyle
         });
     }
 }
@@ -2480,8 +2479,7 @@
         middlePoints.forEach((point, index) => {
             const circle = drawCircle(PlaitBoard.getRoughSVG(this.board), point, LINE_AUTO_COMPLETE_DIAMETER, {
                 stroke: 'none',
-                fill: rgbaToHEX(PRIMARY_COLOR, LINE_AUTO_COMPLETE_OPACITY),
-                fillStyle: 'solid'
+                fill: rgbaToHEX(PRIMARY_COLOR, LINE_AUTO_COMPLETE_OPACITY)
             });
             circle.classList.add(`line-auto-complete-${index}`);
             this.autoCompleteG.appendChild(circle);
@@ -2527,7 +2525,8 @@
         return getEngine(TableSymbols.table).draw(this.board, rectangle, {
             strokeWidth,
             stroke: strokeColor,
-            strokeLineDash
+            strokeLineDash,
+            fillStyle: element.fillStyle
         }, {
             element: element
         });
@@ -4069,7 +4068,7 @@
     draw(board, rectangle, options) {
         const points = getCommentPoints(rectangle);
         const rs = PlaitBoard.getRoughSVG(board);
-        const polygon = rs.polygon(points, { ...options, fillStyle: 'solid' });
+        const polygon = rs.polygon(points, options);
         setStrokeLinecap(polygon, 'round');
         return polygon;
     },
@@ -4116,7 +4115,7 @@
         draw(board, rectangle, options) {
             const points = getPoints(rectangle);
             const rs = PlaitBoard.getRoughSVG(board);
-            const polygon = rs.polygon(points, { ...options, fillStyle: 'solid' });
+            const polygon = rs.polygon(points, options);
             setStrokeLinecap(polygon, 'round');
             return polygon;
         },
@@ -4206,7 +4205,7 @@
         draw(board, rectangle, options) {
             const centerPoint = [rectangle.x + rectangle.width / 2, rectangle.y + rectangle.height / 2];
             const rs = PlaitBoard.getRoughSVG(board);
-            const shape = rs.ellipse(centerPoint[0], centerPoint[1], rectangle.width, rectangle.height, { ...options, fillStyle: 'solid' });
+            const shape = rs.ellipse(centerPoint[0], centerPoint[1], rectangle.width, rectangle.height, options);
             setStrokeLinecap(shape, 'round');
             return shape;
         },
@@ -4431,7 +4430,7 @@
 
 const RectangleEngine = {
     draw(board, rectangle, options) {
-        return drawRectangle(board, rectangle, { ...options, fillStyle: 'solid' });
+        return drawRectangle(board, rectangle, options);
     },
     isInsidePoint(rectangle, point) {
         const rangeRectangle = RectangleClient.getRectangleByPoints([point, point]);
@@ -4455,7 +4454,7 @@
 
 const RoundRectangleEngine = {
     draw(board, rectangle, options) {
-        return drawRoundRectangle(PlaitBoard.getRoughSVG(board), rectangle.x, rectangle.y, rectangle.x + rectangle.width, rectangle.y + rectangle.height, { ...options, fillStyle: 'solid' }, false, getRoundRectangleRadius(rectangle));
+        return drawRoundRectangle(PlaitBoard.getRoughSVG(board), rectangle.x, rectangle.y, rectangle.x + rectangle.width, rectangle.y + rectangle.height, options, false, getRoundRectangleRadius(rectangle));
     },
     isInsidePoint(rectangle, point) {
         return isPointInRoundRectangle(point, rectangle, getRoundRectangleRadius(rectangle));
@@ -4528,7 +4527,7 @@
         const point9 = [x1 + rectangle.width / 4, y2];
         const point10 = [x1 + rectangle.width / 4, rectangle.y + rectangle.height];
         const point11 = [x1 + rectangle.width / 2, y2];
-        const shape = rs.path(`M${point2[0]} ${point2[1]} A ${radius} ${radius}, 0, 0, 1, ${point3[0]} ${point3[1]} L ${point4[0]} ${point4[1]} A ${radius} ${radius}, 0, 0, 1, ${point5[0]} ${point5[1]} L    ${point11[0]} ${point11[1]}  ${point10[0]} ${point10[1]}   ${point9[0]} ${point9[1]}   ${point6[0]} ${point6[1]} A ${radius} ${radius}, 0, 0, 1, ${point7[0]} ${point7[1]} L ${point8[0]} ${point8[1]} A ${radius} ${radius}, 0, 0, 1, ${point1[0]} ${point1[1]} Z`, { ...options, fillStyle: 'solid' });
+        const shape = rs.path(`M${point2[0]} ${point2[1]} A ${radius} ${radius}, 0, 0, 1, ${point3[0]} ${point3[1]} L ${point4[0]} ${point4[1]} A ${radius} ${radius}, 0, 0, 1, ${point5[0]} ${point5[1]} L    ${point11[0]} ${point11[1]}  ${point10[0]} ${point10[1]}   ${point9[0]} ${point9[1]}   ${point6[0]} ${point6[1]} A ${radius} ${radius}, 0, 0, 1, ${point7[0]} ${point7[1]} L ${point8[0]} ${point8[1]} A ${radius} ${radius}, 0, 0, 1, ${point1[0]} ${point1[1]} Z`, options);
         setStrokeLinecap(shape, 'round');
         return shape;
     },
@@ -4680,7 +4679,7 @@
 
 const TerminalEngine = {
     draw(board, rectangle, options) {
-        return drawRoundRectangle(PlaitBoard.getRoughSVG(board), rectangle.x, rectangle.y, rectangle.x + rectangle.width, rectangle.y + rectangle.height, { ...options, fillStyle: 'solid' }, false, getStartEndRadius(rectangle));
+        return drawRoundRectangle(PlaitBoard.getRoughSVG(board), rectangle.x, rectangle.y, rectangle.x + rectangle.width, rectangle.y + rectangle.height, options, false, getStartEndRadius(rectangle));
     },
     isInsidePoint(rectangle, point) {
         return isPointInRoundRectangle(point, rectangle, getStartEndRadius(rectangle));
@@ -4849,7 +4848,7 @@
     draw(board, rectangle, options) {
         const rs = PlaitBoard.getRoughSVG(board);
         const shape = rs.path(`M${rectangle.x} ${rectangle.y} L${rectangle.x + (rectangle.width * 3) / 4} ${rectangle.y} A  ${rectangle.width /
-            4} ${rectangle.height / 2}, 0, 0, 1,${rectangle.x + (rectangle.width * 3) / 4} ${rectangle.y + rectangle.height} L${rectangle.x} ${rectangle.y + rectangle.height} Z`, { ...options, fillStyle: 'solid' });
+            4} ${rectangle.height / 2}, 0, 0, 1,${rectangle.x + (rectangle.width * 3) / 4} ${rectangle.y + rectangle.height} L${rectangle.x} ${rectangle.y + rectangle.height} Z`, options);
         setStrokeLinecap(shape, 'round');
         return shape;
     },
@@ -4890,7 +4889,7 @@
 const StoredDataEngine = {
     draw(board, rectangle, options) {
         const rs = PlaitBoard.getRoughSVG(board);
-        const shape = rs.path(`M${rectangle.x + rectangle.width / 10} ${rectangle.y} L${rectangle.x + rectangle.width} ${rectangle.y} A  ${rectangle.width / 10} ${rectangle.height / 2}, 0, 0, 0,${rectangle.x + rectangle.width} ${rectangle.y + rectangle.height} L${rectangle.x + rectangle.width / 10} ${rectangle.y + rectangle.height}A  ${rectangle.width / 10} ${rectangle.height / 2}, 0, 0, 1,${rectangle.x + rectangle.width / 10} ${rectangle.y}`, { ...options, fillStyle: 'solid' });
+        const shape = rs.path(`M${rectangle.x + rectangle.width / 10} ${rectangle.y} L${rectangle.x + rectangle.width} ${rectangle.y} A  ${rectangle.width / 10} ${rectangle.height / 2}, 0, 0, 0,${rectangle.x + rectangle.width} ${rectangle.y + rectangle.height} L${rectangle.x + rectangle.width / 10} ${rectangle.y + rectangle.height}A  ${rectangle.width / 10} ${rectangle.height / 2}, 0, 0, 1,${rectangle.x + rectangle.width / 10} ${rectangle.y}`, options);
         setStrokeLinecap(shape, 'round');
         return shape;
     },
@@ -4963,7 +4962,7 @@
         const shape = rs.path(`M${rectangle.x} ${rectangle.y} H${rectangle.x + rectangle.width} V${rectangle.y + rectangle.height} H${rectangle.x} Z M${rectangle.x + rectangle.width * 0.06} ${rectangle.y} L${rectangle.x + rectangle.width * 0.06} ${rectangle.y +
             rectangle.height} M${rectangle.x + rectangle.width - rectangle.width * 0.06} ${rectangle.y} L${rectangle.x +
             rectangle.width -
-            rectangle.width * 0.06} ${rectangle.y + rectangle.height}`, { ...options, fillStyle: 'solid' });
+            rectangle.width * 0.06} ${rectangle.y + rectangle.height}`, options);
         setStrokeLinecap(shape, 'round');
         return shape;
     },
@@ -5041,7 +5040,7 @@
         L${rectangle.x + rectangle.width} ${rectangle.y + rectangle.height / 2}
         M${rectangle.x + rectangle.width / 2} ${rectangle.y}
         L${rectangle.x + rectangle.width / 2} ${rectangle.y + rectangle.height}
-        `, { ...options, fillStyle: 'solid' });
+        `, options);
         setStrokeLinecap(shape, 'round');
         return shape;
     }
@@ -5062,7 +5061,7 @@
         L${line1Points[1][0]} ${line1Points[1][1]}
         M${line2Points[0][0]} ${line2Points[0][1]}
         L${line2Points[1][0]} ${line2Points[1][1]}
-        `, { ...options, fillStyle: 'solid' });
+        `, options);
         setStrokeLinecap(shape, 'round');
         return shape;
     }
@@ -5078,7 +5077,7 @@
             (rectangle.height / 9) * 3}, ${rectangle.x + rectangle.width / 2} ${rectangle.y +
             rectangle.height -
             rectangle.height / 9} T${rectangle.x} ${rectangle.y + rectangle.height - rectangle.height / 9}           
-            `, { ...options, fillStyle: 'solid' });
+            `, options);
         setStrokeLinecap(shape, 'round');
         return shape;
     },
@@ -5169,7 +5168,7 @@
                 M${rectangle.x + 5} ${rectangle.y + 10} H${rectangle.x + rectangle.width - 10} V${rectangle.y + rectangle.height - rectangle.height / 9} 
                     
                 M${rectangle.x + 10} ${rectangle.y + 5} H${rectangle.x + rectangle.width - 5} V${rectangle.y + rectangle.height - rectangle.height / 9 - 10 - 3}
-            `, { ...options, fillStyle: 'solid' });
+            `, options);
         setStrokeLinecap(shape, 'round');
         return shape;
     },
@@ -5245,7 +5244,7 @@
             A${rectangle.width / 2} ${rectangle.height * 0.15}, 0, 0, 0,${rectangle.x} ${rectangle.y + rectangle.height * 0.15} 
             V${rectangle.y + rectangle.height - rectangle.height * 0.15}
             A${rectangle.width / 2} ${rectangle.height * 0.15}, 0, 0, 0, ${rectangle.x + rectangle.width} ${rectangle.y + rectangle.height - rectangle.height * 0.15}
-            V${rectangle.y + rectangle.height * 0.15}`, { ...options, fillStyle: 'solid' });
+            V${rectangle.y + rectangle.height * 0.15}`, options);
         setStrokeLinecap(shape, 'round');
         return shape;
     },
@@ -5322,7 +5321,7 @@
             H${rectangle.x + rectangle.width * 0.15}
             A${rectangle.width * 0.15} ${rectangle.height / 2}, 0, 0, 0, ${rectangle.x + rectangle.width * 0.15} ${rectangle.y +
             rectangle.height}
-            H${rectangle.x + rectangle.width - rectangle.width * 0.15}`, { ...options, fillStyle: 'solid' });
+            H${rectangle.x + rectangle.width - rectangle.width * 0.15}`, options);
         setStrokeLinecap(shape, 'round');
         return shape;
     },
@@ -5398,7 +5397,7 @@
         const shape = rs.path(`M${rectangle.x} ${rectangle.y} h${rectangle.width} v${rectangle.height} h${-rectangle.width} v${-rectangle.height}
             M${rectangle.x} ${rectangle.y + rectangle.height / 10} h${rectangle.width}
             M${rectangle.x + rectangle.width / 10} ${rectangle.y} v${rectangle.height}
-            `, { ...options, fillStyle: 'solid' });
+            `, options);
         setStrokeLinecap(shape, 'round');
         return shape;
     },
@@ -5466,7 +5465,7 @@
             ${lowerCurve.controlPoint2[0]} ${lowerCurve.controlPoint2[1]},
             ${lowerCurve.endPoint[0]} ${lowerCurve.endPoint[1]}`
         ].join(' ');
-        const shape = rs.path(pathData, { ...options, fillStyle: 'solid', fill: 'transparent' });
+        const shape = rs.path(pathData, { ...options, fill: 'transparent' });
         setStrokeLinecap(shape, 'round');
         return shape;
     },
@@ -5539,7 +5538,7 @@
             ${lowerCurve.controlPoint2[0]} ${lowerCurve.controlPoint2[1]},
             ${lowerCurve.endPoint[0]} ${lowerCurve.endPoint[1]}`
         ].join(' ');
-        const shape = rs.path(pathData, { ...options, fillStyle: 'solid', fill: 'transparent' });
+        const shape = rs.path(pathData, { ...options, fill: 'transparent' });
         setStrokeLinecap(shape, 'round');
         return shape;
     },
@@ -5592,7 +5591,7 @@
         const rs = PlaitBoard.getRoughSVG(board);
         const shape = rs.path(`M${rectangle.x + rectangle.width * 0.075} ${rectangle.y + rectangle.height} H${rectangle.x} V${rectangle.y} H${rectangle.x +
             rectangle.width * 0.075}
-           `, { ...options, fillStyle: 'solid', fill: 'transparent' });
+           `, { ...options, fill: 'transparent' });
         setStrokeLinecap(shape, 'round');
         return shape;
     },
@@ -5639,7 +5638,7 @@
             H${rectangle.x + rectangle.width * 0.15}
             L${rectangle.x} ${rectangle.y + rectangle.height / 2}
             Z
-            `, { ...options, fillStyle: 'solid' });
+            `, options);
         setStrokeLinecap(shape, 'round');
         return shape;
     },
@@ -5701,7 +5700,7 @@
                         y: y + ACTIVE_STROKE_WIDTH,
                         width: width - ACTIVE_STROKE_WIDTH * 2,
                         height: height - ACTIVE_STROKE_WIDTH * 2
-                    }, { fill: cell.fill, fillStyle: 'solid', strokeWidth: 0 });
+                    }, { fill: cell.fill, fillStyle: roughOptions?.fillStyle, strokeWidth: 0 });
                     const cellRightBorder = drawLine(rs, [x + width, y], [x + width, y + height], roughOptions);
                     const cellBottomBorder = drawLine(rs, [x, y + height], [x + width, y + height], roughOptions);
                     g.append(cellRectangle, cellRightBorder, cellBottomBorder);
@@ -5846,7 +5845,7 @@
             `M${leftLegLine[0][0]} ${leftLegLine[0][1]} L${leftLegLine[1][0]} ${leftLegLine[1][1]}`,
             `M${rightLegLine[0][0]} ${rightLegLine[0][1]} L${rightLegLine[1][0]} ${rightLegLine[1][1]}`
         ].join(' ');
-        const shape = rs.path(pathData, { ...options, fillStyle: 'solid' });
+        const shape = rs.path(pathData, options);
         setStrokeLinecap(shape, 'round');
         return shape;
     },
@@ -5918,7 +5917,7 @@
 const ContainerEngine = {
     draw(board, rectangle, options) {
         const rs = PlaitBoard.getRoughSVG(board);
-        const shape = rs.path(`M${rectangle.x} ${rectangle.y} H${rectangle.x + rectangle.width} V${rectangle.y + rectangle.height} H${rectangle.x} Z M${rectangle.x + 40} ${rectangle.y} L${rectangle.x + 40} ${rectangle.y + rectangle.height} `, { ...options, fillStyle: 'solid' });
+        const shape = rs.path(`M${rectangle.x} ${rectangle.y} H${rectangle.x + rectangle.width} V${rectangle.y + rectangle.height} H${rectangle.x} Z M${rectangle.x + 40} ${rectangle.y} L${rectangle.x + 40} ${rectangle.y + rectangle.height} `, options);
         setStrokeLinecap(shape, 'round');
         return shape;
     },
@@ -5989,7 +5988,7 @@
             `V${points.leftMiddle[1]}`,
             `H${points.middlePoint[0]}`
         ].join(' ');
-        const shape = rs.path(pathData, { ...options, fillStyle: 'solid' });
+        const shape = rs.path(pathData, options);
         setStrokeLinecap(shape, 'round');
         return shape;
     },
@@ -6069,7 +6068,7 @@
             H${rectangle.x + rectangle.width / 3 - 8}
             L${rectangle.x + rectangle.width / 3} ${rectangle.y + 16}
             V${rectangle.y}
-            `, { ...options, fillStyle: 'solid' });
+            `, options);
         setStrokeLinecap(shape, 'round');
         return shape;
     },
@@ -6135,7 +6134,7 @@
     draw(board, rectangle, options) {
         const rs = PlaitBoard.getRoughSVG(board);
         const lines = getDeletionLines(rectangle);
-        const shape = rs.path(lines.map(([from, to]) => `M${from[0]} ${from[1]} L${to[0]} ${to[1]}`).join(' '), { ...options, fillStyle: 'solid', strokeWidth: 4 });
+        const shape = rs.path(lines.map(([from, to]) => `M${from[0]} ${from[1]} L${to[0]} ${to[1]}`).join(' '), { ...options, strokeWidth: 4 });
         setStrokeLinecap(shape, 'round');
         return shape;
     },
@@ -6176,7 +6175,7 @@
         const shape = rs.path(`M${rectangle.x} ${rectangle.y} H${rectangle.x + rectangle.width} V${rectangle.y + rectangle.height} H${rectangle.x} Z M${rectangle.x + rectangle.width * 0.125} ${rectangle.y} L${rectangle.x + rectangle.width * 0.125} ${rectangle.y +
             rectangle.height} M${rectangle.x + rectangle.width - rectangle.width * 0.125} ${rectangle.y} L${rectangle.x +
             rectangle.width -
-            rectangle.width * 0.125} ${rectangle.y + rectangle.height}`, { ...options, fillStyle: 'solid' });
+            rectangle.width * 0.125} ${rectangle.y + rectangle.height}`, options);
         setStrokeLinecap(shape, 'round');
         return shape;
     },
@@ -6225,7 +6224,7 @@
             Z
             M${rectangle.x + rectangle.width - 16} ${rectangle.y} 
             A16 16, 0,0,1, ${rectangle.x + rectangle.width} ${rectangle.y + 16}
-            `, { ...options, fillStyle: 'solid' });
+            `, options);
         setStrokeLinecap(shape, 'round');
         return shape;
     },
@@ -6321,10 +6320,7 @@
             // 最后一条线
             `M${line2[0][0]} ${line2[0][1]} H${line2[1][0]}`
         ].join(' ');
-        const shape = rs.path(pathData, {
-            ...options,
-            fillStyle: 'solid'
-        });
+        const shape = rs.path(pathData, options);
         setStrokeLinecap(shape, 'round');
         return shape;
     },
@@ -6404,7 +6400,6 @@
         ].join(' ');
         const shape = rs.path(pathData, {
             ...options,
-            fillStyle: 'solid',
             fill: 'transparent'
         });
         setStrokeLinecap(shape, 'round');
@@ -6495,10 +6490,7 @@
             `H${line.endX}`,
             ...arcCommands.map((command) => `A${command.rx} ${command.ry} ${command.xAxisRotation} ${command.largeArcFlag} ${command.sweepFlag} ${command.endX} ${command.endY}`)
         ].join(' ');
-        const shape = rs.path(pathData, {
-            ...options,
-            fillStyle: 'solid'
-        });
+        const shape = rs.path(pathData, options);
         setStrokeLinecap(shape, 'round');
         return shape;
     },
@@ -6586,7 +6578,7 @@
             `M${points.bottomBoxEnd[0]} ${points.bottomBoxEnd[1]}`,
             `V${points.mainEnd[1]}`
         ].join(' ');
-        const shape = rs.path(pathData, { ...options, fillStyle: 'solid' });
+        const shape = rs.path(pathData, options);
         setStrokeLinecap(shape, 'round');
         return shape;
     },
@@ -6669,7 +6661,7 @@
             V${rectangle.y + rectangle.height} 
             H${rectangle.x} Z
 
-            `, { ...options, fillStyle: 'solid' });
+            `, options);
         const componentShape = ComponentEngine.draw(board, componentRectangle, options);
         shape.append(componentShape);
         setStrokeLinecap(shape, 'round');
@@ -6714,7 +6706,6 @@
         const rs = PlaitBoard.getRoughSVG(board);
         return drawRoundRectangle(rs, rectangle.x, rectangle.y, rectangle.x + rectangle.width, rectangle.y + rectangle.height, {
             ...options,
-            fillStyle: 'solid',
             dashGap: 10,
             strokeLineDash: [10, 10]
         }, false, 4);
@@ -8014,8 +8005,7 @@
                     const circleG = drawCircle(PlaitBoard.getRoughSVG(board), ref.connectorPoint || ref.edgePoint, 6, {
                         stroke: SELECTION_BORDER_COLOR,
                         strokeWidth: SNAPPING_STROKE_WIDTH,
-                        fill: SELECTION_BORDER_COLOR,
-                        fillStyle: 'solid'
+                        fill: SELECTION_BORDER_COLOR
                     });
                     boundShapeG.appendChild(circleG);
                 }
@@ -8266,8 +8256,7 @@
                 // function 1: dnd
                 reactionG = drawCircle(PlaitBoard.getRoughSVG(board), hitPoint, LINE_AUTO_COMPLETE_HOVERED_DIAMETER, {
                     stroke: 'none',
-                    fill: rgbaToHEX(PRIMARY_COLOR, LINE_AUTO_COMPLETE_HOVERED_OPACITY),
-                    fillStyle: 'solid'
+                    fill: rgbaToHEX(PRIMARY_COLOR, LINE_AUTO_COMPLETE_HOVERED_OPACITY)
                 });
                 PlaitBoard.getActiveHost(board).append(reactionG);
                 PlaitBoard.getBoardContainer(board).classList.add(CursorClass.crosshair);
